### DISCUSSION TITLE
Входящие с GOIP-4 не проходят

### DISCUSSION THREAD

**Discussion by shesrst** [mikopbx/Core#75] [Q&A] [answered]
Добрый день!
Прошу помочь решить проблему входящих звонков, поступающих на GSM шлюз GOIP-4.
Настроил по инструкции: https://wiki.mikopbx.com/faq:goip
Исходящие работают, а входящие нет: на мобильном телефоне просто сбрасывается вызов, а если смотреть через SSH, что происходит при звонке (asterisk -rvvvvv), то при входящем звонке вообще ничего не происходит.
В указанной выше инструкции указано, что в настройках шлюза в разделе Call In в поле CID Forward Mode нужно указать значение Use CID as SIP Caller ID.
Заметил, что если поставить значение Disable или Use Remote Party ID, то входящий проходит, но номер звонящего не определяется.
Заранее благодарю.

**Comment by boffart** [comment, marked as answer]
Тут первым делом я бы посмотрел есть ли **INVITE** на Mikopbx (от шлюза). 
[Снятие лога с помощью Sngrep](https://wiki.mikopbx.com/faq:sngrep)

Дале посмотрел бы в [логи](https://wiki.mikopbx.com/system-diagnostic). Тут интересен файл **asterisk/messages**
В нем поставьте фильтр по строке "**public-direct**". 

Я предполагаю, что вызов входящий до АТС доходит, но без авторизации. В том случае шлюз должен получить в ответ на **INVITE** сообщение not found. А в логе будет ошибка, что в контексте **public-direct-dial** не найден какой то EXTEN

**Comment by shesrst** [comment]
Прошу прощения за неответ. Работа вынудила отвлечься от этого мероприятия.
Да, вы правы так и есть. В Sngrep вижу, что звонок приходит, но он  REJECTED, т.к.  Not Found.
Я так понял, что это из-за того, что номер мобильно телефона определяется с +7. 
 В asterisk/messages вижу:
Call from 'anonymous' (UDP:192.168.0.40:5060) to extension '2200100' rejected because extension not found in context 'public-direct-dial'.
Дальше, думаю, разберусь. Спасибо )

**Reply by boffart** [reply]
В итоге задачу можно свести к описанию "[входящего маршрута](https://wiki.mikopbx.com/incoming-routes)" **без** указания "**провайдера**", но с описанием **DID** номера, что виден в логах "**2200100**".
